pyccel/ast/c_concepts.py

```diff
@@ -34,3 +35,149 @@ def obj(self):
         """The object whose address is of interest
         """
         return self._obj
+
+#------------------------------------------------------------------------------
+class CStringExpression(Basic):
+    """
+    Internal class used to hold a C string that has LiteralStrings and C macros
+    Parameters:
+        *args: str or LiteralString or CMacro or CStringExpression
+            any number of arguments to be added to the expression
+            note: they will get added in the order provided
+    Example:
+        CStringExpression(
+            CMacro("m"),
+            CStringExpression(
+                LiteralString("the macro is: "),
+                CMacro("mc")
+            ),
+            LiteralString("."),
+        )
+    """
+    __slots__  = ('_expression',)
+    _attribute_nodes  = ()
+
+    def __init__(self, *args):
+        super().__init__()
+        self._expression = []
+        for arg in args:
+            self += arg
+
+    def __repr__(self):
+        return ''.join([repr(e) for e in self._expression])
+
+    def __str__(self):
+        return ''.join([str(e) for e in self._expression])
+
+    def __add__(self, o):
+        """
+        append the argument o to the end of the list _expression
+        Parameter:
+            o: str or LiteralString or CMacro or CStringExpression
+        """
+        if isinstance(o, str):
+            o = LiteralString(o)
+        if not isinstance(o, (LiteralString, CMacro, CStringExpression)):
+            raise TypeError(f"unsupported operand type(s) for +: '{self.__class__}' and '{type(o)}'")
+        self.expression.append(o)
+        return self
+
+    def __radd__(self, o):
+        if isinstance(o, LiteralString):
+            return CStringExpression(o, self)
+        return NotImplemented
+
+    def intersperse(self, lst):
+        """
+        insert self between each element of the list `lst`
+        Parameter:
+            lst: list of (str or LiteralString or CMacro or CStringExpression)
+            the list to insert self between its elements
+        Example:
+            a = [
+                CMacro("m"),
+                CStringExpression(LiteralString("the macro is: ")),
+                LiteralString("."),
+            ]
+
+            b = CStringExpression("?").intersperse(a)
+
+            is the same as:
+
+            b = CStringExpression(
+                CMacro("m"),
+                CStringExpression("?"),
+                CStringExpression(LiteralString("the macro is: ")),
+                CStringExpression("?"),
+                LiteralString("."),
+            )
+        """
+        result = CStringExpression()
+        if not lst:
+            return result
+        result += lst[0]
+        for elm in lst[1:]:
+            result += self
+            result += elm
+        return result
+
+    def get_flat_expression_list(self):
+        """
+        returns an list of LiteralStrings and CMacros after merging every
```

Comment:
typo: `returns a list of...`

Reply:
done

Reply:
[024460f](https://github.com/pyccel/pyccel/pull/1181/commits/024460f409713f343bebaf106e9b7399c5b1f7d0)